src/python/pants/base/exiter.py

```diff
@@ -70,6 +71,9 @@ def exit(self, result=PANTS_SUCCEEDED_EXIT_CODE, msg=None, out=None):
       except Exception as e:
         # If the file is already closed, or any other error occurs, just log it and continue to
         # exit.
+        if msg:
+          logger.warning("Encountered error when trying to log this message: {}".format(msg))
+          traceback.print_stack()
```

Comment:
I'm not sure where this traceback prints, could you confirm that it's to a good place?

Reply:
For pantsd, it will probably print nowhere. For nopantsd it will print to `sys.stderr`. Given this is only logged when we've failed to write to `out`, I don't think there's a better behaviour we can offer...